Documentation "Configuring systemd-journald" lists incorrect steps

mburke5678 opened the pull request:
According to @richm _there are a few problems with that doc_, including a step to restart Rsyslog, which is not in 4.x.
https://github.com/openshift/openshift-docs/pull/16597#discussion_r323850430

Also: https://bugzilla.redhat.com/show_bug.cgi?id=1751289

I removed the topic from 4.2 and 4.1. This PR is to fix the topic and place it back. 
PR removing for 4.1: https://github.com/openshift/openshift-docs/pull/16676

vikram-redhat commented:
@mburke5678 this in for 4.3?

mburke5678 commented:
@richm @anpingli Richard suggested that there were problems with the `cluster-logging-systemd-scaling.adoc `. Can you take a look at the topic and suggest changes? 
I removed the topic until we can correct it. 
Thank you!!

anpingli commented:
@mburke5678  There are both rhcos and RHEL nodes.  That is OK to modify the RHEL nodes directly.  @mffiedler @richm   Shall we use same configuration for both?

mffiedler commented:
Since rsyslog will not graduate to GA in 4.3, should be just leave the topic out?

richm commented:
> Since rsyslog will not graduate to GA in 4.3, should be just leave the topic out?

This topic isn't specific to rsyslog - it will affect how fluentd reads journald logs as well.

mburke5678 commented:
@anpingli I have a BZ from a TAM suggesting to make the changes to the MachineConfig 

_Should list details for creating edits via MachineConfig for both worker and master nodes._
https://bugzilla.redhat.com/show_bug.cgi?id=1751289#c0

mburke5678 commented:
@alanconway @jcantrill Awhile back, @richm pointed out that topic is wrong in the docs. It never got updated for 4.x. 

I have a separate BZ from TAM [1] that suggests the systemd parameters for rate limiting and such go into a MachineSet, rather than the `/etc/systemd/journald.conf` file as in 3.x. I see a `systemd.units.contents` section in the MachineCongifs and added the parameters there:

    systemd:
      Compress: 'no'
      MaxRetentionSec: 30s
      RateLimitBurst: 10000
      RateLimitInterval: 1s
      Storage: volatile
      units:

Also, I found this list of systemd parameters and defaults [2] from an older verion and updated the docs to include them using definitions from here [3].

Can you help verify if any of this is correct for 4.x?

Michael 

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1751289
[2]https://github.com/openshift/openshift-ansible/pull/3753/files#diff-40b7a7231e77d95ca6009dc9bcc0f470R30-R41
[3} https://www.freedesktop.org/software/systemd/man/journald.conf.html

mburke5678 commented:
@jcantrill Can you help me move this one along? Been lingering for a while.

mburke5678 commented:
@jcantrill Can you help me move this one along? Been lingering for a while.
Preview:
http://file.rdu.redhat.com/~mburke/logging-fix-systemd/logging/config/cluster-logging-systemd.html

richm commented:
> Create a journald.conf file with the required settings:

This looks like YAML format, while journald.conf is in INI format https://www.freedesktop.org/software/systemd/man/journald.conf.html

In OCP 3.11, you would use that YAML format to specify all of these journald parameters when deploying using openshift-ansible https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/openshift_node/defaults/main.yml#L108
not sure if this is documented already elsewhere

mburke5678 commented:
@anpingli PTAL

anpingli commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -2,38 +2,124 @@
+        verification: {}
+      filesystem: root
+      mode: 644
+      path: journald.conf <3>
```
The mco won't work, filed bug https://bugzilla.redhat.com/show_bug.cgi?id=1801016.

anpingli replied:
It should be /etc/systemd/journald.conf

anpingli commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -2,38 +2,124 @@
-Storage=volatile
-Compress=no
-MaxRetentionSec=30s
+
```
@RH-ematysek  have you any suggestion about the values?

mburke5678 replied:
@RH-ematysek as Anping asked, do we want to document all of the parameters I listed in Lines 24-34 above?

mburke5678 commented:
@anpingli I updated the doc to change the `mode` value to `420` as suggested in the BZ [1] and added the wording from the BZ. 
See: https://github.com/openshift/openshift-docs/pull/16662/commits/4c3080add1fbb707b87e7f8daf2ffd5c1e5979d7

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1801016#c1

mburke5678 commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,127 @@
+        source: data:text/plain;charset=utf-8;base64,${jrnl_cnf}
+        verification: {}
+      filesystem: root
+      mode: 420 <1>
```
@anpingli Thank you for pointing this out. I did not see that you updated that BZ with this information.

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
 // Module included in the following assemblies:
 //
-// * logging/cluster-logging-deploy.adoc
+// * logging/congig/cluster-logging-systemd
```
config :)

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
-. Update to *systemd-219-22.el7.x86_64*.
+. Create a `journald.conf` file with the required settings:
++
+[source,yaml]
```
Highlighting weird, because this isn't yaml. Suggest removing this.

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
++
+<1> Specify whether you want logs compressed before they are written to the file system. 
+Specify `yes` to compress the message or `no` to not compress. Default is `yes`.
+<2> Configure whether to forward log messages. Defaults to `no` for each. Specify
```
Maybe add a colon after Specify?

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+* `ForwardToConsole` to forward logs to the system console.
+* `ForwardToKsmg` to forward logs to the kernel log buffer.
+* `ForwardToSyslog` to forward to a syslog daemon.
+* `ForwardToWall` to forward messages to user's wall.
```
to a user's wall?

mburke5678 replied:
Console. 
Changed to "sent as wall messages to all logged-in users"

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+* `ForwardToKsmg` to forward logs to the kernel log buffer.
+* `ForwardToSyslog` to forward to a syslog daemon.
+* `ForwardToWall` to forward messages to user's wall.
+<3> Specify the maximum time to store journal entries. Enter digits to specify seconds. Or 
```
Suggest backticks around 0.

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+* `ForwardToWall` to forward messages to user's wall.
+<3> Specify the maximum time to store journal entries. Enter digits to specify seconds. Or 
+include a unit: "year", "month", "week", "day", "h" or "m". Enter 0 to disable. Default is `1month`. 
+<4> Congifure rate limiting. If, during the time interval defined by `RateLimitIntervalSec`, more logs than specified in `RateLimitBurst` 
```
Configure

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+<3> Specify the maximum time to store journal entries. Enter digits to specify seconds. Or 
+include a unit: "year", "month", "week", "day", "h" or "m". Enter 0 to disable. Default is `1month`. 
+<4> Congifure rate limiting. If, during the time interval defined by `RateLimitIntervalSec`, more logs than specified in `RateLimitBurst` 
+are received, all further messages within the interval are dropped until the interval is over. We recommend setting 
```
We don't say "we". Suggest just "It is recommended.."

mburke5678 replied:
It is awful... ;-)

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+<4> Congifure rate limiting. If, during the time interval defined by `RateLimitIntervalSec`, more logs than specified in `RateLimitBurst` 
+are received, all further messages within the interval are dropped until the interval is over. We recommend setting 
+`RateLimitInterval=1s` and `RateLimitBurst=10000`, which are the defaults.
+<5> Specify how logs are stored. Default is `persistent`: 
```
The default

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
++
+[NOTE]
+====
+If you removing the rate limit, you might see increased CPU utilization on the
```
If you removing -> If you are removing

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
-Compress=no
-MaxRetentionSec=30s
+
+....
```
Only 3 dots for continuation

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+----
 
-. Add the following to the *_/etc/systemd/journald.conf_* file:
+. Create a new MachineConfig for master and/or worker and add the `journal.conf` parameters:
```
reword to not do and/or

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+      path: /etc/systemd/journald.conf <2>
+  systemd: {}
 ----
+<1> Set the permissions for the `journal.conf` file. We recommend setting to `0644` permissions. 
```
Same about removing "we"

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
-After removing the rate limit, you might see increased CPU utilization on the
-system logging daemons as it processes any messages that would have previously
-been throttled.
+. Use `oc describe machineconfigpool/worker` to monitor the status of the rollout of the new rendered configuration to each node:
```
Not critical, but suggest removing the command from the sentence (you don't do that in the steps above). Just

```
Monitor the status of the rollout of the new rendered configuration to each node:
```

bergerhoffer commented on modules/cluster-logging-systemd-scaling.adoc:
```diff
@@ -1,39 +1,126 @@
+API Version:  machineconfiguration.openshift.io/v1
+Kind:         MachineConfigPool
+
+....
```
3 dots

bergerhoffer commented on modules/nodes-nodes-rebooting-affinity.adoc:
```diff
@@ -50,4 +50,4 @@ This example assumes the container image registry pod has a label of
 
 . Enable the `*MatchInterPodAffinity*` scheduler predicate in the scheduling policy file. 
-For more information, see xref:../../nodes/scheduling/nodes-scheduler-default.adoc#nodes-scheduler-default[Configuring the Default Scheduler].
+For more information, see Configuring the Default Scheduler.
```
I don't think this is a good idea to leave a non-xref dependency like this (specifying the title but it not being an xref). Because even though it's not technically dependent, it doesn't make sense w/o the other module. I don't know how others have been handling it, but maybe you could say something like "For more information, see the documentation on configuring the default scheduler"? I don't know, wdyt?

mburke5678 replied:
That topic doesn't belong in here! Will remove and consider your question in context!

mburke5678 commented:
/cherrypick enterprise-4.1

mburke5678 commented:
/cherrypick enterprise-4.2

mburke5678 commented:
/cherrypick enterprise-4.3

mburke5678 commented:
/cherrypick enterprise-4.4
